src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -119,84 +138,174 @@ interface Props {
+                  values.transactions.length === 1 && (
+                    <div className={styles.transactionTitle}>
+                      1.{' '}
+                      <FormattedMessage
+                        {...ConstantsMSG[
+                          values.transactions[index].transactionType
+                        ]}
+                      />
+                    </div>
+                  )}
+                <DetailsItem
+                  label={MSG.safe}
+                  value={
+                    <AddressDetailsView
+                      item={(values.safe as never) as AnyUser}
+                      isSafeItem
+                    />
+                  }
                 />
-              }
-            />
-          )}
-          {transactionDetails?.map(({ key, label, value }) => (
-            <DetailsItem
-              key={key}
-              label={label}
-              value={value(values.transactions[0][key], token)}
-            />
-          ))}
+                {values.transactions[index].transactionType !==
+                  TransactionTypes.CONTRACT_INTERACTION && (
+                  <DetailsItem
+                    label={MSG.to}
+                    value={
+                      <AddressDetailsView
+                        item={
+                          (values.transactions[index]
+                            .recipient as never) as AnyUser
+                        }
+                        isSafeItem={false}
+                      />
+                    }
+                  />
+                )}
+                {values.transactions[index].transactionType !== '' &&
+                  transactionTypeFieldsMap[
+                    values.transactions[index].transactionType
+                  ].map(({ key, label, value }, idx, arr) => (
+                    <DetailsItem
+                      key={key}
+                      label={label}
+                      value={value(
+                        values.transactions[index][key],
+                        tokens[index],
+                      )}
+                      isLastItem={idx + 1 === arr.length}
+                    />
+                  ))}
+              </div>
+            </DialogSection>
+          </div>
         </div>
-      </DialogSection>
-      <DialogSection appearance={{ theme: 'sidePadding' }}>
-        <Input
-          appearance={{ colorSchema: 'grey', theme: 'fat' }}
-          label={MSG.transactionsTitle}
-          name="transactionsTitle"
-          placeholder=""
-        />
-      </DialogSection>
-      <DialogSection>
-        <Annotations label={MSG.description} name="annotation" />
-      </DialogSection>
+      ))}
+      <div className={styles.footer}>
+        <DialogSection appearance={{ theme: 'sidePadding' }}>
+          <Input
+            appearance={{ colorSchema: 'grey', theme: 'fat' }}
+            label={MSG.transactionsTitle}
+            name="transactionsTitle"
+            placeholder=""
```

Comment:
I know that you didn't introduce this but can you remove this empty placeholder? :) I don't think it's needed